### DISCUSSION TITLE
failing test case related to 'SLURM_LOCALID'

### DISCUSSION THREAD

**Discussion by tkonopka** [Lightning-AI/pytorch-lightning#12551] [General]
Hi all. I am trying to set up pytorch-lightning for the first time, following `tests/README.md` to run the test suite, and experiencing a minor glitch. I would like to ask for feedback if this is something to follow up or not.

Running `make test` on a local machine gives one failed test: `test_default_attributes` in `tests/plugins/environments/test_slurm_environments.py`. It fails [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/plugins/environments/test_slurm_environment.py#L36-L38) because it eventually expects `os.environ["SLURM_LOCALID"]` to throw a KeyError, but 'SLURM_LOCALID' turns out to be defined in the environment and KeyError does not happen. 

The same test case passes when triggered through `python -m pytest tests/`.

I found two ways to avoid the error:
1. remove [this line](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/Makefile#L4) from the Makefile
2. set `clear=True` in the decorator for the specific test case [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/plugins/environments/test_slurm_environment.py#L23). i.e. replace that line with `@mock.patch.dict(os.environ, {}, clear=True)`

I am not sure if these are reasonable in the big picture. Your thoughts would be appreciated: is this something to handle locally only, or could be mentioned in the README, or is it a bug?

Many thanks

**Comment by akihironitta** [comment]
Hi @tkonopka! Thanks for reporting this!

As you suggest, it should definitely be set `patch.dict(..., clear=True)` so that the default env variable value doesn't affect how the test case behaves. Would it be possible for you to send a fix PR?

**Reply by tkonopka** [reply]
Sure, I'll prepare a PR. Thanks for looking into this.

**Reply by akihironitta** [reply]
Thanks again for your contribution in #12561!
